pkgs/applications/video/obs-studio/obs-websocket.nix

```diff
@@ -0,0 +1,51 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchpatch
+, cmake
+, qtbase
+, obs-studio
+, asio_1_10
+, websocketpp
+}:
+
+stdenv.mkDerivation rec {
+  pname = "obs-websocket";
+  version = "4.9.0";
+
+  src = fetchFromGitHub {
+    owner = "Palakis";
+    repo = "obs-websocket";
+    rev = version;
+    sha256 = "1dxih5czcfs1vczbq48784jvmgs8awbsrwk8mdfi4pg8n577cr1w";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ qtbase obs-studio asio_1_10 websocketpp ];
+
+  dontWrapQtApps = true;
+
+  cmakeFlags = with lib; [
+    "-DLIBOBS_INCLUDE_DIR=${obs-studio.src}/libobs"
+  ];
+
+  # obs-studio expects the shared object to be located in bin/32bit or bin/64bit
+  # https://github.com/obsproject/obs-studio/blob/d60c736cb0ec0491013293c8a483d3a6573165cb/libobs/obs-nix.c#L48
+  postInstall = let
+    pluginPath = {
+      i686-linux = "bin/32bit";
+      x86_64-linux = "bin/64bit";
+    }.${stdenv.targetPlatform.system} or (throw "Unsupported system: ${stdenv.targetPlatform.system}");
+  in ''
+    mkdir -p $out/share/obs/obs-plugins/obs-websocket/${pluginPath}
+    ln -s $out/lib/obs-plugins/obs-websocket.so $out/share/obs/obs-plugins/obs-websocket/${pluginPath}
+  '';
+
+  meta = with lib; {
+    description = "Remote-control OBS Studio through WebSockets";
+    homepage = "https://github.com/Palakis/obs-websocket";
+    maintainers = with maintainers; [ erdnaxe ];
+    license = licenses.gpl2;
```

Comment:
gpl2Only or gpl2Plus?

Reply:
From the LICENSE:
>    This program is free software; you can redistribute it and/or modify
>    it under the terms of the GNU General Public License as published by
>    the Free Software Foundation; either version 2 of the License, or
>    (at your option) any later version.

So I guess it's gpl2Plus?

Reply:
Right; the "at your option any later version" clause means it's gpl2Plus.